src/components/SingleReview.jsx

```diff
@@ -0,0 +1,63 @@
+import { React, useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getSingleReview } from "../utils/api";
+import moment from "moment";
+
+function SingleReview() {
+  const { review_id } = useParams();
+  const [singleReview, setSingleReview] = useState({});
+
+  useEffect(() => {
+    getSingleReview(review_id).then((singleReviewFromApi) => {
+      setSingleReview(singleReviewFromApi);
+    });
+  }, [review_id, singleReview]);
```

Comment:
this will cause a loop if singleReview is in the dependency array and setSingleReview is being used in the useEffect.  This will work perfectly if we remove singleReview from the dependency array